src/pages/home/Home.jsx

```diff
@@ -1,34 +1,69 @@
 import "./home.css";
 import { useDocTitle } from "../../hooks/useDocTitle";
-import {
-  LeftSideBar,
-  Loader,
-  Post,
-  RightSideBar,
-  ScrollToTop,
-} from "../../components";
+import { InfinitySpin as Loader } from "react-loader-spinner";
+import { LeftSideBar, Post, RightSideBar, ScrollToTop } from "../../components";
 import { Link } from "react-router-dom";
 import { useDispatch, useSelector } from "react-redux";
 import { MdImage } from "react-icons/md/";
+import { AiOutlineCloseCircle } from "react-icons/ai/";
 import { useState } from "react";
 import { toast } from "react-toastify";
 import { createPost, sortByValue } from "../../redux/reducers/postsSlice";
+import { sortPosts } from "../../hooks/sortPosts";
 
 export function Home() {
   useDocTitle("Home - Smasher - Manoj Sarna");
-  const [postDetails, setPostDetails] = useState("");
+  const [postDetails, setPostDetails] = useState({
+    content: "",
+    postImage: "",
+  });
+
   const [selectSortValue, setSelectSortValue] = useState("Latest");
+  const loading = useSelector((state) => state.posts.loading);
+
+  const [isLoading, setIsLoading] = useState(false);
   const user = useSelector((state) => state.auth.user);
   const dispatch = useDispatch();
   const posts = useSelector((state) => state.posts.posts);
   let sortBy = useSelector((state) => state.posts.sortBy);
```

Comment:
suggestion: you can destructure `posts` and `sortBy` from `state.posts` directly